### PR TITLE
Add support for cohorts metadata in Af1 -- alimanfoo suggestions 2023-01-26

### PR DESCRIPTION
**Pull request by alimanfoo**
Hi @leehart, putting some suggestions here on your PR at https://github.com/malariagen/malariagen-data-python/pull/318

* [x] Move loading of JSON config file to parent class.
* [x] Move `snp_allele_counts()` to parent class.
* [x] Move `snp_allele_frequencies()` to parent class.
* [x] Move `results_cache_get()` and `results_cache_set()` to parent class.
* [x] Move `fst_gwss()` to parent class, and remove subclass overrides for related plotting functions.
* [x] Tidy up `wgs_data_catalog()`, no need to override in Ag3.
* [x] Move `view_alignments()` to parent class.

### CODE REVIEW

File: malariagen_data/ag3.py [hunk 1]
```diff
@@ -174,26 +174,13 @@ def __init__(
...
-        # load config.json
-        path = f"{self._base_path}/v3-config.json"
-        with self._fs.open(path) as f:
-            self._config = json.load(f)
-
-        if cohorts_analysis is None:
-            self._cohorts_analysis = self._config["DEFAULT_COHORTS_ANALYSIS"]
-        else:
-            self._cohorts_analysis = cohorts_analysis
-
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:186]: Move to parent class.

File: malariagen_data/af1.py [hunk 2]
```diff
@@ -142,21 +144,6 @@ def __init__(
...
-        if cohorts_analysis is None:
-            self._cohorts_analysis = self._config["DEFAULT_COHORTS_ANALYSIS"]
-        else:
-            self._cohorts_analysis = cohorts_analysis
-
-        if site_filters_analysis is None:
-            self._site_filters_analysis = self._config["DEFAULT_SITE_FILTERS_ANALYSIS"]
-        else:
-            self._site_filters_analysis = site_filters_analysis
-
```
Comment (alimanfoo) [inline malariagen_data/af1.py:159]: Move to parent class.

File: malariagen_data/ag3.py [hunk 3]
```diff
@@ -174,26 +174,13 @@ def __init__(
...
         if species_analysis is None:
             self._species_analysis = self._config["DEFAULT_SPECIES_ANALYSIS"]
         else:
             self._species_analysis = species_analysis
 
-        if site_filters_analysis is None:
-            self._site_filters_analysis = self._config["DEFAULT_SITE_FILTERS_ANALYSIS"]
-        else:
-            self._site_filters_analysis = site_filters_analysis
-
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:196]: Move to parent class.

File: malariagen_data/anopheles.py [hunk 4]
```diff
@@ -260,11 +282,6 @@ def _public_releases(self):
     def _site_annotations_zarr_path(self):
         raise NotImplementedError("Must override _site_annotations_zarr_path")
 
-    @property
-    @abstractmethod
-    def _cohorts_analysis(self):
-        raise NotImplementedError("Must override _cohorts_analysis")
-
```
Comment (alimanfoo) [inline malariagen_data/anopheles.py:267]: Changing this to be an instance attribute.

**Comment by leehart** [comment]
Thanks @alimanfoo 👍

**Comment by alimanfoo** [comment]
Hi @leehart, I think I'm done here, nothing else I can see to refactor for now. Ready for review.

**Comment by leehart** [comment]
Awesome! Thanks @alimanfoo 👍 👍

**Comment by alimanfoo** [comment]
Cool, thanks Lee.

**Pull request merged**